File: leetcode124.py
# Definition for a binary tree node.
import sys
class TreeNode:
     def __init__(self, val=0, left=None, right=None):
         self.val = val
         self.left = left
         self.right = right
class Solution:
    def maxPathSum(self, root: Optional[TreeNode]) -> int:
        self.res = -sys.maxsize-1

        def dfs(node):
            if node == None:
                return 0
            left, right = max(0,dfs(node.left)), max(dfs(node.right),0)
            self.res= max(self.res,node.val+left+right)
            return node.val+max(left,right)

        dfs(root)
        return self.res

File: c++/.vs/c++/leetcode124.cpp

 //Definition for a binary tree node.
 struct TreeNode {
      int val;
      TreeNode *left;
      TreeNode *right;
      TreeNode() : val(0), left(nullptr), right(nullptr) {}
      TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
      TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
  };
 
class Solution {
public:
    int res = INT_MIN;
    int generate(TreeNode* node) {
        if (node == nullptr) return 0;
        int left = max(0, generate(node->left)), right = max(0, generate(node->right));
        res = max(res, node->val + left + right);
        return node->val + max(left, right);
    }
    int maxPathSum(TreeNode* root) {
        generate(root);
        return res;
    }
};